Does not compile on x86_64

By default does not get compiled on x86_64 (debian). The compilation error is as follows:

```
$ make
*--------------------------------------------------
*  Process Control Daemon
*  Build date: Sun, 08 Oct 2017 10:28:02 +0530 :
*--------------------------------------------------
Checking write permission: /tmp/1
Checking write permission: /tmp/1/
Checking write permission: /tmp/1/
Building PCD...
make[1]: Entering directory '/tmp/pcd/ipc/src'
  CC [C] 	ipc.o
  LINK	 	libipc
make[1]: Leaving directory '/tmp/pcd/ipc/src'
make[1]: Entering directory '/tmp/pcd/pcd/src/pcdapi/src'
  CC [C] 	pcdapi.o
/tmp/pcd/pcd/src/pcdapi/src/pcdapi.c: In function ‘PCD_api_malloc_and_send’:
/tmp/pcd/pcd/src/pcdapi/src/pcdapi.c:113:67: warning: comparison between ‘IPC_status_e’ and ‘enum <anonymous>’ [-Wenum-compare]
     if ( IPC_get_context_by_owner( &pcdCtx, CONFIG_PCD_OWNER_ID ) != PCD_STATUS_OK )
                                                                   ^
/tmp/pcd/pcd/src/pcdapi/src/pcdapi.c:122:47: warning: comparison between ‘IPC_status_e’ and ‘enum <anonymous>’ [-Wenum-compare]
     if ( IPC_start( pcdClient, &pcdTmpCtx, 0) != PCD_STATUS_OK )
                                               ^
  LINK	 	libpcd
make[1]: Leaving directory '/tmp/pcd/pcd/src/pcdapi/src'
make[1]: Entering directory '/tmp/pcd/pcd/src'
  CC [C] 	condchk.o
  CC [C] 	errlog.o
  CC [C] 	except.o
/tmp/pcd/pcd/src/except.c: In function ‘PCD_dump_maps_file’:
/tmp/pcd/pcd/src/except.c:167:10: warning: variable ‘i’ set but not used [-Wunused-but-set-variable]
  int32_t i;
          ^
  CC [C] 	failact.o
  CC [C] 	main.o
  CC [C] 	misc.o
  CC [C] 	parser.o
  CC [C] 	pcd_api.o
/tmp/pcd/pcd/src/pcd_api.c: In function ‘PCD_api_init’:
/tmp/pcd/pcd/src/pcd_api.c:113:24: warning: comparison between ‘IPC_status_e’ and ‘enum <anonymous>’ [-Wenum-compare]
     if ( IPC_init( 0 ) != PCD_STATUS_OK )
                        ^
/tmp/pcd/pcd/src/pcd_api.c:121:62: warning: comparison between ‘IPC_status_e’ and ‘enum <anonymous>’ [-Wenum-compare]
     if ( IPC_start( CONFIG_PCD_SERVER_NAME, &pcdContext, 0 ) != PCD_STATUS_OK )
                                                              ^
/tmp/pcd/pcd/src/pcd_api.c:129:59: warning: comparison between ‘IPC_status_e’ and ‘enum <anonymous>’ [-Wenum-compare]
     if ( IPC_set_owner( pcdContext, CONFIG_PCD_OWNER_ID ) != PCD_STATUS_OK )
                                                           ^
/tmp/pcd/pcd/src/pcd_api.c: In function ‘PCD_api_check_messages’:
/tmp/pcd/pcd/src/pcd_api.c:165:54: warning: comparison between ‘IPC_status_e’ and ‘enum <anonymous>’ [-Wenum-compare]
         if ( IPC_get_msg_context( msg, &msgContext ) == PCD_STATUS_OK )
                                                      ^
/tmp/pcd/pcd/src/pcd_api.c:327:49: warning: comparison between ‘IPC_status_e’ and ‘enum <anonymous>’ [-Wenum-compare]
             if ( IPC_reply_msg( msg, replyMsg ) != PCD_STATUS_OK )
                                                 ^
/tmp/pcd/pcd/src/pcd_api.c: In function ‘PCD_api_reply_message’:
/tmp/pcd/pcd/src/pcd_api.c:361:50: warning: comparison between ‘IPC_status_e’ and ‘enum <anonymous>’ [-Wenum-compare]
     if ( IPC_get_msg_context( msg, &msgContext ) == PCD_STATUS_OK )
                                                  ^
/tmp/pcd/pcd/src/pcd_api.c:385:45: warning: comparison between ‘IPC_status_e’ and ‘enum <anonymous>’ [-Wenum-compare]
         if ( IPC_reply_msg( msg, replyMsg ) != PCD_STATUS_OK )
                                             ^
  CC [C] 	process.o
/tmp/pcd/pcd/src/process.c: In function ‘PCD_process_spawn’:
/tmp/pcd/pcd/src/process.c:506:21: warning: implicit declaration of function ‘setpriority’ [-Wimplicit-function-declaration]
                     setpriority( PRIO_PROCESS, 0, rule->sched.niceSched );
                     ^
/tmp/pcd/pcd/src/process.c:506:34: error: ‘PRIO_PROCESS’ undeclared (first use in this function)
                     setpriority( PRIO_PROCESS, 0, rule->sched.niceSched );
                                  ^
/tmp/pcd/pcd/src/process.c:506:34: note: each undeclared identifier is reported only once for each function it appears in
/tmp/pcd/pcd/src/process.c: In function ‘PCD_process_reboot’:
/tmp/pcd/pcd/src/process.c:1014:17: warning: variable ‘i’ set but not used [-Wunused-but-set-variable]
         int32_t i;
                 ^
Makefile:58: recipe for target 'process.o' failed
make[1]: *** [process.o] Error 1
make[1]: Leaving directory '/tmp/pcd/pcd/src'
Makefile:141: recipe for target 'pcd' failed
make: *** [pcd] Error 2
```

--- a/pcd/src/except.c
+++ b/pcd/src/except.c
@@ -164,7 +164,6 @@ static void PCD_dump_maps_file( pid_t pid )
     struct stat fbuf;
     char mapsFile[ 22 ];
     int32_t fd;
-	int32_t i;
 
     sprintf( mapsFile, "%s/%d.maps", CONFIG_PCD_TEMP_PATH, pid );
 
@@ -179,12 +178,12 @@ static void PCD_dump_maps_file( pid_t pid )
         char buffer[ 512 ];
         int32_t readBytes = 0;
 
-        i = write( STDERR_FILENO, "\nMaps file:\n\n", 13 );
+        write( STDERR_FILENO, "\nMaps file:\n\n", 13 );
 
         /* Read the maps file and display it on the console */
         while ( ( readBytes = read( fd, buffer, sizeof( buffer ) ) ) > 0 )
         {
-            i = write( STDERR_FILENO, buffer, readBytes );
+            write( STDERR_FILENO, buffer, readBytes );
         }
 
         close( fd );

--- a/pcd/src/pcd_api.c
+++ b/pcd/src/pcd_api.c
@@ -110,7 +110,7 @@ static PCD_status_e PCD_api_signal( rule_t *rule, int32_t sig )
 PCD_status_e PCD_api_init( void )
 {
     /* Init IPC */
-    if ( IPC_init( 0 ) != PCD_STATUS_OK )
+    if ( IPC_init( 0 ) != IPC_STATUS_OK )
     {
         PCD_PRINTF_STDERR( "Failed to init PCD API");
 
@@ -118,7 +118,7 @@ PCD_status_e PCD_api_init( void )
     }
 
     /* Start IPC */
-    if ( IPC_start( CONFIG_PCD_SERVER_NAME, &pcdContext, 0 ) != PCD_STATUS_OK )
+    if ( IPC_start( CONFIG_PCD_SERVER_NAME, &pcdContext, 0 ) != IPC_STATUS_OK )
     {
         PCD_PRINTF_STDERR( "Failed to start IPC" );
 
@@ -126,7 +126,7 @@ PCD_status_e PCD_api_init( void )
     }
 
     /* Set owner */
-    if ( IPC_set_owner( pcdContext, CONFIG_PCD_OWNER_ID ) != PCD_STATUS_OK )
+    if ( IPC_set_owner( pcdContext, CONFIG_PCD_OWNER_ID ) != IPC_STATUS_OK )
     {
         IPC_stop( pcdContext );
 
@@ -162,7 +162,7 @@ PCD_status_e PCD_api_check_messages( void )
         IPC_context_t msgContext;
 
         /* Check if we need to reply */
-        if ( IPC_get_msg_context( msg, &msgContext ) == PCD_STATUS_OK )
+        if ( IPC_get_msg_context( msg, &msgContext ) == IPC_STATUS_OK )
         {
             /* Allocate memory for reply message */
             replyMsg = IPC_alloc_msg( pcdContext, sizeof( pcdApiReplyMessage_t ) );
@@ -324,7 +324,7 @@ PCD_status_e PCD_api_check_messages( void )
             replyData->retval = retval;
 
             /* Send response */
-            if ( IPC_reply_msg( msg, replyMsg ) != PCD_STATUS_OK )
+            if ( IPC_reply_msg( msg, replyMsg ) != IPC_STATUS_OK )
             {
                 IPC_free_msg( replyMsg );
             }
@@ -358,7 +358,7 @@ void PCD_api_reply_message( void *cookie, PCD_status_e retval )
     data = IPC_get_msg( msg );
 
     /* Check if we need to reply */
-    if ( IPC_get_msg_context( msg, &msgContext ) == PCD_STATUS_OK )
+    if ( IPC_get_msg_context( msg, &msgContext ) == IPC_STATUS_OK )
     {
         /* Allocate memory for reply message */
         replyMsg = IPC_alloc_msg( pcdContext, sizeof( pcdApiReplyMessage_t ) );
@@ -382,7 +382,7 @@ void PCD_api_reply_message( void *cookie, PCD_status_e retval )
         replyData->retval = retval;
 
         /* Send response */
-        if ( IPC_reply_msg( msg, replyMsg ) != PCD_STATUS_OK )
+        if ( IPC_reply_msg( msg, replyMsg ) != IPC_STATUS_OK )
         {
             PCD_PRINTF_STDERR( "Failed to send termination reply message" );
             IPC_free_msg( replyMsg );

--- a/pcd/src/pcdapi/src/pcdapi.c
+++ b/pcd/src/pcdapi/src/pcdapi.c
@@ -110,7 +110,7 @@ static PCD_status_e PCD_api_malloc_and_send( const struct ruleId_t *ruleId, pcdA
         pcdApiInitDone = True;
     }
 
-    if ( IPC_get_context_by_owner( &pcdCtx, CONFIG_PCD_OWNER_ID ) != PCD_STATUS_OK )
+    if ( IPC_get_context_by_owner( &pcdCtx, CONFIG_PCD_OWNER_ID ) != IPC_STATUS_OK )
     {
         printf( "pcd: Error: Failed to find PCD context\n");       
         return PCD_STATUS_INVALID_RULE;
@@ -119,7 +119,7 @@ static PCD_status_e PCD_api_malloc_and_send( const struct ruleId_t *ruleId, pcdA
     sprintf( pcdClient, CONFIG_PCD_CLIENTS_NAME_PREFIX "%d", getpid() );
 
     /* Create temporary destination point */
-    if ( IPC_start( pcdClient, &pcdTmpCtx, 0) != PCD_STATUS_OK )
+    if ( IPC_start( pcdClient, &pcdTmpCtx, 0) != IPC_STATUS_OK )
     {
         printf( "pcd: Error: Failed to start IPC\n");
         return PCD_STATUS_NOK;

--- a/pcd/src/process.c
+++ b/pcd/src/process.c
@@ -62,6 +62,8 @@
 #include "pcd_api.h"
 #include "ipc.h"
 
+#include "sys/resource.h"
+
 /**************************************************************************/
 /*      LOCAL DEFINITIONS AND VARIABLES                                   */
 /**************************************************************************/
@@ -1011,10 +1013,9 @@ void PCD_process_reboot( void )
     else
     {
         const char msg[]= "pcd: Reboot disabled in debug mode, exiting.\n";
-        int32_t i;
 		
         /* Display reboot message */
-        i = write( STDERR_FILENO, msg, sizeof(msg) );
+        write( STDERR_FILENO, msg, sizeof(msg) );
 
         /* Exit abnormally */
         exit( 1 );

